src/content/React-CSharp/ClientApp/src/components/api-authorization/Logout.js

```diff
@@ -1,128 +1,113 @@
+          case AuthenticationResultStatus.Redirect:
+            break;
+          case AuthenticationResultStatus.Success:
+            await navigateToReturnUrl(returnUrl);
+            break;
+          case AuthenticationResultStatus.Fail:
+            setMessage(result.message);
+            break;
+          default:
+            throw new Error("Invalid authentication result status.");
         }
-    }
-
-    async logout(returnUrl) {
-        const state = { returnUrl };
-        const isauthenticated = await authService.isAuthenticated();
-        if (isauthenticated) {
-            const result = await authService.signOut(state);
-            switch (result.status) {
-                case AuthenticationResultStatus.Redirect:
-                    break;
-                case AuthenticationResultStatus.Success:
-                    await this.navigateToReturnUrl(returnUrl);
-                    break;
-                case AuthenticationResultStatus.Fail:
-                    this.setState({ message: result.message });
-                    break;
-                default:
-                    throw new Error("Invalid authentication result status.");
-            }
+      } else {
+        setMessage("You successfully logged out!");
+      }
+    };
+    
+    const action = props.action;
+    switch (action) {
+      case LogoutActions.Logout:
+        if (!!window.history.state.state.local) {
+          logout(getReturnUrl());
         } else {
-            this.setState({ message: "You successfully logged out!" });
+          // This prevents regular links to <app>/authentication/logout from triggering a logout
+          setIsReady(true);
+          setMessage("The logout was not initiated from within the page.");
         }
+        break;
+      case LogoutActions.LogoutCallback:
+        processLogoutCallback();
+        break;
+      case LogoutActions.LoggedOut:
+        setIsReady(true);
+        setMessage("You successfully logged out!");
+        break;
+      default:
+        throw new Error(`Invalid action '${action}'`);
     }
 
-    async processLogoutCallback() {
-        const url = window.location.href;
-        const result = await authService.completeSignOut(url);
-        switch (result.status) {
-            case AuthenticationResultStatus.Redirect:
-                // There should not be any redirects as the only time completeAuthentication finishes
-                // is when we are doing a redirect sign in flow.
-                throw new Error('Should not redirect.');
-            case AuthenticationResultStatus.Success:
-                await this.navigateToReturnUrl(this.getReturnUrl(result.state));
-                break;
-            case AuthenticationResultStatus.Fail:
-                this.setState({ message: result.message });
-                break;
-            default:
-                throw new Error("Invalid authentication result status.");
-        }
-    }
+    setIsReady(true);
+  }, [props])
```

Comment:
`props` shouldn't be here

```suggestion
  }, [])
```